File: stripe/src/test/java/com/stripe/android/model/SourceReceiverTest.java
package com.stripe.android.model;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;

import java.util.Objects;

import static org.junit.Assert.assertEquals;

/**
 * Test class for {@link SourceReceiver}.
 */
public class SourceReceiverTest {

    private static final String EXAMPLE_JSON_RECEIVER = "{" +
            "\"address\": \"test_1MBhWS3uv4ynCfQXF3xQjJkzFPukr4K56N\"," +
            "\"amount_charged\": 10," +
            "\"amount_received\": 20," +
            "\"amount_returned\": 30" +
            "}";

    @Test
    public void fromJson_createsExpectedObject() throws JSONException {
        final SourceReceiver sourceReceiver = Objects.requireNonNull(
                        SourceReceiver.fromJson(new JSONObject(EXAMPLE_JSON_RECEIVER)));
        assertEquals("test_1MBhWS3uv4ynCfQXF3xQjJkzFPukr4K56N",
                sourceReceiver.getAddress());
        assertEquals(10, sourceReceiver.getAmountCharged());
        assertEquals(20, sourceReceiver.getAmountReceived());
        assertEquals(30, sourceReceiver.getAmountReturned());
    }
}

File: stripe/src/test/java/com/stripe/android/model/SourceReceiverTest.kt
package com.stripe.android.model

import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * Test class for [SourceReceiver].
 */
class SourceReceiverTest {

    @Test
    fun fromJson_createsExpectedObject() {
        val sourceReceiver = SourceFixtures.SOURCE_RECEIVER
        assertEquals("test_1MBhWS3uv4ynCfQXF3xQjJkzFPukr4K56N", sourceReceiver.address)
        assertEquals(10, sourceReceiver.amountCharged)
        assertEquals(20, sourceReceiver.amountReceived)
        assertEquals(30, sourceReceiver.amountReturned)
    }
}
